x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/agent_upgrade_status.test.tsx

```diff
@@ -0,0 +1,185 @@
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_REQUESTED',
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade requested');
+    });
+
+    it('should render UPG_SCHEDULED state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_SCHEDULED',
+          metadata: {
+            scheduled_at: '2023-08-09T10:11:12Z',
+          },
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade scheduled');
+    });
+
+    it('should render UPG_DOWNLOADING state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_DOWNLOADING',
+          metadata: {
+            download_percent: 16.4,
+          },
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade downloading');
+    });
+
+    it('should render UPG_EXTRACTING state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_EXTRACTING',
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade extracting');
+    });
+
+    it('should render UPG_REPLACING state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_REPLACING',
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade replacing');
+    });
+
+    it('should render UPG_RESTARTING state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_RESTARTING',
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade restarting');
+    });
+
+    it('should render UPG_WATCHING state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_WATCHING',
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade monitoring');
+    });
+
+    it('should render UPG_ROLLBACK state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_ROLLBACK',
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade rolled back');
+    });
+
+    it('should render UPG_FAILED state correctly', () => {
+      const results = render({
+        agentUpgradeDetails: {
+          target_version: 'XXX',
+          action_id: 'xxx',
+          state: 'UPG_FAILED',
+          metadata: {
+            failed_state: 'UPG_DOWNLOADING',
+            error_msg: 'Something went wrong',
+          },
+        },
+      });
+      expectBadgeLabel(results, 'Upgrade failed');
```

Comment:
nit: the tooltip text could be verified for cases where there is a text with variables